src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -413,45 +400,45 @@ public static void TestWithGuardMethods()
             {
                 Target.UnsupportedInWindows();
                 Target.UnsupportedInWindows10();
-                [|Target.UnsupportedOnBrowser()|]; // row 15 expected diagnostic - browser unsupported
-                [|Target.UnsupportedOnWindowsAndBrowser()|]; // expected diagnostic - browser unsupported
-                [|Target.UnsupportedOnWindows10AndBrowser()|]; // expected diagnostic - browser unsupported
+                [|Target.UnsupportedOnBrowser()|]; // row 15 'Target.UnsupportedOnBrowser()' is unsupported on: 'browser'. This call site is reachable on all platforms.
+                [|Target.UnsupportedOnWindowsAndBrowser()|]; // 'Target.UnsupportedOnWindowsAndBrowser()' is unsupported on: 'browser'. This call site is reachable on all platforms.
+                [|Target.UnsupportedOnWindows10AndBrowser()|]; // 'Target.UnsupportedOnWindows10AndBrowser()' is unsupported on: 'browser'. This call site is reachable on all platforms.
             }
 
             if (!OperatingSystemHelper.IsWindowsVersionAtLeast(10))
             {
-                [|Target.UnsupportedInWindows()|]; // row 22 expected diagnostic - windows unsupported
+                [|Target.UnsupportedInWindows()|]; // 'Target.UnsupportedInWindows()' is unsupported on: 'windows'. This call site is reachable on all platforms.
                 Target.UnsupportedInWindows10();
-                [|Target.UnsupportedOnBrowser()|]; // expected diagnostic - browser unsupported
-                [|Target.UnsupportedOnWindowsAndBrowser()|]; // expected 2 diagnostics - windows and browser unsupported
+                [|Target.UnsupportedOnBrowser()|]; // 'Target.UnsupportedOnBrowser()' is unsupported on: 'browser'. This call site is reachable on all platforms.
+                [|Target.UnsupportedOnWindowsAndBrowser()|]; // 'Target.UnsupportedOnWindowsAndBrowser()' is unsupported on: 'windows', 'browser'. This call site is reachable on all platforms.
                 [|Target.UnsupportedOnWindows10AndBrowser()|]; // expected diagnostic - browser unsupported
             }
 
             if (OperatingSystemHelper.IsWindows())
             {
-                [|Target.UnsupportedInWindows()|]; // row 31 expected diagnostic - windows unsupported
+                [|Target.UnsupportedInWindows()|]; // 'Target.UnsupportedInWindows()' is unsupported on: 'windows'. This call site is reachable on: 'Windows'.
                 [|Target.UnsupportedInWindows10()|]; // expected diagnostic - windows 10 unsupported
                 Target.UnsupportedOnBrowser();
                 [|Target.UnsupportedOnWindowsAndBrowser()|]; // expected diagnostic - windows unsupported
-                [|Target.UnsupportedOnWindows10AndBrowser()|]; // expected diagnostic - windows 10 unsupported
+                [|Target.UnsupportedOnWindows10AndBrowser()|]; // 'Target.UnsupportedOnWindows10AndBrowser()' is unsupported on: 'windows' 10.0 and later. This call site is reachable on: 'Windows'.
```

Comment:
> 'Target.UnsupportedOnWindows10AndBrowser()' is unsupported on: 'windows' 10.0 and later. This call site is reachable on: 'Windows'.

In this example, it might not be clear that the call site is reachable on all versions of Windows, since that isn't called out explicitly.  And the casing of 'windows' vs. 'Windows' could lead a user to think there's a case sensitivity problem causing the error instead of the all versions aspects.

What would you think about the following?

1. Always lower-casing the platforms
2. Using a message that includes "(all versions)" when that makes a difference

Reply:
> 1. Always lower-casing the platforms

Platforms are printed how they provided in the attribute, i think not all platform names suitable for lower casing (should be the same as how we use in TFM), further, we could address platform name casing with the other analyzer which would warn for platform name/versions correctness and attributes consistency

> 2. Using a message that includes "(all versions)" when that makes a difference

I will give it try

Reply:
> not all platform names suitable for lower casing

Makes sense.  I presume when we detect the code is within a guard, we're getting the platform name from the `Is<Platform>` method, just stripping off the `Is`, which means we'll always be proper cased ("Windows").  The first character is likely where we'll see the difference.

Drawing attention to the version mismatch is going to be the key then.  We should consider including in the documentation a note that platform names are case-insensitive, in case anyone gets that impression and goes to the docs to check if that's the case.

Reply:
> I presume when we detect the code is within a guard, we're getting the platform name from the Is<Platform> method, just stripping off the Is, which means we'll always be proper cased ("Windows"). The first character is likely where we'll see the difference.

Oh yeah in this case it is produced from the guard methods, that is why it is capital

> Drawing attention to the version mismatch is going to be the key then. We should consider including in the documentation a note that platform names are case-insensitive, in case anyone gets that impression and goes to the docs to check if that's the case.

Good point